extensions/wikia/NjordPrototype/NjordController.class.php

```diff
@@ -0,0 +1,325 @@
+<?php
+
+class NjordController extends WikiaController {
+
+	const HERO_IMAGE_FILENAME = 'wikia-hero-image';
+	const THUMBNAILER_SIZE_SUFIX = '1600px-0';
+
+	const MAINPAGE_PAGE = 'mainpage';
+
+	private static $wikiDataModel = null;
+
+	public function __construct() {
+		parent::__construct();
+
+		$this->initWikiDataModel();
+	}
+
+	private function initWikiDataModel() {
+		if (self::$wikiDataModel === null) {
+			self::$wikiDataModel = new WikiDataModel(Title::newMainPage()->getText());
+			self::$wikiDataModel->getFromProps();
+		}
+	}
+
+	public function getWikiMarkup() {
+		$articleWikiMarkup = '';
+		$articleTitle = $this->getRequest()->getVal( 'articleTitle' );
+		$pageTitleObj = Title::newFromText( $articleTitle );
+		if ( isset($pageTitleObj) && $pageTitleObj->exists() ) {
+			$pageArticleObj = new Article( $pageTitleObj );
+			$articleWikiMarkup = $pageArticleObj->getPage()->getText();
+		}
+		$this->getResponse()->setFormat( 'json' );
+		$this->getResponse()->setVal( 'wikiMarkup', $articleWikiMarkup );
+	}
+
+	public function MainPageModuleSave() {
+		$articleTitle = $this->getRequest()->getVal( 'pageTitle' );
+		$wikiMarkup = $this->getRequest()->getVal( 'wikiMarkup' );
+		$pageTitleObj = Title::newFromText( $articleTitle );
+		$pageArticleObj = new Article( $pageTitleObj );
+
+		$pageArticleObj->doEdit( $wikiMarkup, '' );
+		$pageArticleObj->doPurge();
+		$newHtml = $pageArticleObj->getPage()->getParserOutput( new ParserOptions( null, null ) )->getText();
+
+		$this->getResponse()->setFormat( 'json' );
+		$this->getResponse()->setVal( 'html', $newHtml );
+	}
+
+	public function index() {
+		global $wgTitle, $wgUser, $wgBlankImgUrl;
+		$wikiDataModel = self::$wikiDataModel;
+
+		$this->wg->SupressPageTitle = true;
+
+		$this->wg->out->addStyle( AssetsManager::getInstance()->getSassCommonURL( 'extensions/wikia/NjordPrototype/css/Njord.scss' ) );
+
+		$wikiDataModel->imageSet = true;
+		if ( !isset( $wikiDataModel->imagePath ) ) {
+			$wikiDataModel->imageSet = false;
+			$wikiDataModel->imagePath = $wgBlankImgUrl;
+			$wikiDataModel->originalImagePath = $wgBlankImgUrl;
+		}
+
+		// template vars
+		$this->wikiData = $wikiDataModel;
+		$this->isAllowedToEdit = $this->wg->user->isAllowed( 'njordeditmode' );
+	}
+
+	public function summary() {
+		global $wgUser;
+		$wikiDataModel = self::$wikiDataModel;
+
+		if ( !$wgUser->isLoggedIn() && $wikiDataModel->isEmpty() ) {
+			return $this->skipRendering();
+		}
+
+		// template vars
+		$this->wikiData = $wikiDataModel;
+		$this->isAllowedToEdit = $this->wg->user->isAllowed( 'njordeditmode' );
+	}
+
+	public function modula() {
+		$this->content = $this->getRequest()->getVal( 'content' );
+		$this->align = $this->getRequest()->getVal( 'align' );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "align" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)